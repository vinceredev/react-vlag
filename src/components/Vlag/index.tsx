import React from "react";
import { useVlag } from "../../libs/context";
interface VlagProps {
  name: string;
  children: (isActive: boolean) => React.ReactElement;
}

const Vlag: React.FC<VlagProps> = (props) => {
  const ff = useVlag();
  return props.children(ff.isEnabled(props.name));
};

export default Vlag;
