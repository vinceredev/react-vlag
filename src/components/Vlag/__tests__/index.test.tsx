import React from 'react';
import renderer from 'react-test-renderer';
import Vlag from '../index';

it('success: renders correctly', () => {
  const tree = renderer
    .create(
      <Vlag name="checkout_page">
        {(isActive) => (isActive ? <>beta checkout</> : <>release checkout</>)}
      </Vlag>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
