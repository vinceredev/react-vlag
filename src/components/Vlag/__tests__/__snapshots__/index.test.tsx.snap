// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`success: renders correctly 1`] = `"release checkout"`;
